com.microsoft.mrtk.graphicstools.unity/Runtime/Experimental/Magnifier/Scripts/MagnifierManager.cs

```diff
@@ -172,7 +178,27 @@ public void ApplyMagnification()
         {
             Shader.SetGlobalFloat(MagnificationPropertyName, 1.0f - Magnification);
         }
-
+        public void GetMousePos()
+        {
+            if (inLensMode)
+            {
+                Vector3 wldpos = Camera.main.ScreenToViewportPoint(Input.mousePosition);
+                Vector4 Mousewldpos = new Vector4(wldpos.x, wldpos.y, 0, 0);
+                Shader.SetGlobalVector("_MousePos", Mousewldpos);
```

Comment:
Rather than call `Shader.SetGlobalVector` with a string for the first parameter could you cache a shader ID? This will eliminate some per frame allocations and lookups.

For example make a member like this:
`private static readonly int mousePosID= Shader.PropertyToID("_MousePos");`

then you can do:
`Shader.SetGlobalVector(mousePosID, Mousewldpos);`